stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/AttachmentsPickerSystemTabFactory.kt

```diff
@@ -56,23 +61,45 @@ import androidx.compose.ui.unit.dp
 import androidx.compose.ui.window.Dialog
 import androidx.compose.ui.window.DialogProperties
 import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.accompanist.permissions.shouldShowRationale
 import io.getstream.chat.android.compose.R
 import io.getstream.chat.android.compose.state.messages.attachments.AttachmentPickerItemState
 import io.getstream.chat.android.compose.state.messages.attachments.AttachmentsPickerMode
 import io.getstream.chat.android.compose.state.messages.attachments.Files
-import io.getstream.chat.android.compose.state.messages.attachments.MediaCapture
-import io.getstream.chat.android.compose.state.messages.attachments.Poll
 import io.getstream.chat.android.compose.ui.theme.ChatTheme
 import io.getstream.chat.android.compose.ui.util.StorageHelperWrapper
+import io.getstream.chat.android.ui.common.contract.internal.CaptureMediaContract
 import io.getstream.chat.android.ui.common.helper.internal.AttachmentFilter
 import io.getstream.chat.android.ui.common.helper.internal.StorageHelper
 import io.getstream.chat.android.ui.common.state.messages.composer.AttachmentMetaData
+import io.getstream.chat.android.ui.common.utils.isPermissionDeclared
+import java.io.File
 
 /**
  * Holds the information required to add support for "files" tab in the attachment picker.
+ *
+ * @param filesAllowed If the option to pick files is included in the attachments picker.
+ * @param mediaAllowed If the option to pick media (images/videos) is included in the attachments picker.
+ * @param captureImageAllowed If the option to capture an image is included in the attachments picker.
+ * @param captureVideoAllowed If the option to capture a video is included in the attachments picker.
+ * @param pollAllowed If the option to create a poll is included in the attachments picker.
  */
-public class AttachmentsPickerSystemTabFactory(private val otherFactories: List<AttachmentsPickerTabFactory>) :
-    AttachmentsPickerTabFactory {
+public class AttachmentsPickerSystemTabFactory(
+    private val filesAllowed: Boolean,
+    private val mediaAllowed: Boolean,
+    private val captureImageAllowed: Boolean,
+    private val captureVideoAllowed: Boolean,
+    private val pollAllowed: Boolean,
+) : AttachmentsPickerTabFactory {
```

Comment:
This seems to be a braking change as well. Would be nice leave backward compatibility and maybe mark it as `@Deprecated` 👍

Reply:
Addressed here: 20746a0e3ce1f61fcc8419ffd601ac1fcb669145

The current constructor is left as it was, but is now deprecated in favour of the newly added one. (Same goes for the `AttachmentsPickerTabFactories` default methods.)